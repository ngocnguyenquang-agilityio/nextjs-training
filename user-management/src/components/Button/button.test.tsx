import { fireEvent, render, screen } from '@testing-library/react';

// Components
import { Button } from '.';

describe('Button Component', () => {
  test('Should render successfully', () => {
    const comp = render(<Button>Button</Button>);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<Button>Button</Button>);
    expect(comp).toMatchSnapshot();
  });

  test('Should render correct variants secondary', () => {
    render(<Button variant="secondary">Button</Button>);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toMatch(
      /bg-gray-100 text-gray-600 transition-colors hover:bg-gray-200 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-500 active:bg-gray-300/gi
    );
  });

  test('Should render correct variants danger ', () => {
    render(<Button variant="danger">Button</Button>);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toMatch(
      /bg-red-500 text-white transition-colors hover:bg-red-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-red-500 active:bg-red-600/gi
    );
  });

  test('Should render correct size sm', () => {
    render(<Button size="sm">Button</Button>);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toMatch(/px-2 py-1/gi);
  });

  test('Should render correct size lg', () => {
    render(<Button size="lg">Button</Button>);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toMatch(/px-8 py-3/gi);
  });

  test('Should render correct rounded sm', () => {
    render(<Button rounded="sm">Button</Button>);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toMatch(/rounded/gi);
  });

  test('Should render correct rounded lg', () => {
    render(<Button rounded="lg">Button</Button>);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toMatch(/rounded-lg/gi);
  });

  test('Should render correct rounded full', () => {
    render(<Button rounded="full">Button</Button>);
    const button = screen.getByRole('button');
    expect(button.getAttribute('class')).toMatch(/rounded-full/gi);
  });

  test('Should call onClick function', () => {
    const mockOnClick = jest.fn();
    render(<Button onClick={mockOnClick}>Button</Button>);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockOnClick.mock.calls.length).toBe(1);
  });
});
