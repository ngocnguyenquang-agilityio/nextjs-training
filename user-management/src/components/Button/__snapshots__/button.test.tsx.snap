// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="flex items-center text-sm font-md bg-blue-500 text-white border border-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
      >
        Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="flex items-center text-sm font-md bg-blue-500 text-white border border-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
    >
      Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
